buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/license/RemoteLicenseFetcher.groovy

```diff
@@ -14,155 +14,159 @@
 
 package com.google.firebase.gradle.plugins.license
 
+import java.io.IOException
 import org.jsoup.Jsoup
 import org.jsoup.examples.HtmlToPlainText
+import org.jsoup.nodes.Document
 
 /**
  * Parse licenses from remote urls*/
-interface RemoteLicenseFetcher extends Serializable {
-  static final TEXT_FORMATTER = new HtmlToPlainText()
+class RemoteLicenseFetcher implements Serializable {
+  private static final HtmlToPlainText TEXT_FORMATTER = new HtmlToPlainText()
 
-  URI getServiceUri()
+  private final String remoteUrl
 
-  String get()
-
-  static final class AndroidSdkTermsFetcher implements RemoteLicenseFetcher {
-    private URI ANDROID_SDK_TERMS_URI = URI.create("https://developer.android.com/studio/terms.html")
+  protected RemoteLicenseFetcher(String remoteUrl) {
+    this.remoteUrl = remoteUrl
+  }
 
-    @Override
-    URI getServiceUri() {
-      ANDROID_SDK_TERMS_URI
-    }
+  public final String getRemoteUrl() {
+    return remoteUrl
+  }
 
-    @Override
-    String get() {
-      def doc = Jsoup.connect(ANDROID_SDK_TERMS_URI.toString()).get()
+  /**
+   * Downloads and returns the text of the license.
+   *
+   * <p>This method will try at most three times to download the license. Failures will be silently
+   * ignored if the license can be successfully downloaded on the second or third attempt. This
+   * method performs a simple, linear backoff in the case of failures to improve chances of
+   * successful connections.
+   */
+  public final String getText() {
+    IOException storedEx = null
+
+    for (int i = 0; i < 3; ++i) {
+      try {
+        if (i > 0) {
+          Thread.sleep(i * 1000)
+        }
+
+        return processDocument(Jsoup.connect(remoteUrl).get())
+      } catch (IOException ex) {
+        if (storedEx == null) {
+          storedEx = ex
+        } else {
+          storedEx.addSuppressed(ex)
+        }
+      }
+    }
+
+    throw storedEx
+  }
 
-      TEXT_FORMATTER.getPlainText(doc.select('#body-content > div.jd-descr > div')[0])
-    }
+  /** Extracts the license text from the rest of the document. */
+  String processDocument(Document doc) {
+    return TEXT_FORMATTER.getPlainText(doc)
   }
 
-  static final class Apache2LicenseFetcher implements RemoteLicenseFetcher {
-    private URI APACHE_2_LICENSE_URI = URI.create("http://www.apache.org/licenses/LICENSE-2.0.txt")
+  static final class AndroidSdkTermsFetcher extends RemoteLicenseFetcher {
```

Comment:
Is it proper to have these Fetchers live in their own files, instead of being a nested static class under `RemoteLicenseFetcher`?

Reply:
I think the nested classes are fine here.

Reply:
OK. I just feel the full name could be a little confusing, since it is now `<the-package-name>.RemoteLicenseFetcher.AndroidSdkTermsFetcher`. But it is totally up to you.